rand_os/src/lib.rs

```diff
@@ -84,12 +82,14 @@ impl RngCore for OsRng {
     }
 
     fn try_fill_bytes(&mut self, dest: &mut [u8]) -> Result<(), Error> {
-        // Some systems do not support reading 0 random bytes.
-        // (And why waste a system call?)
-        if dest.len() == 0 { return Ok(()); }
-        
-        getrandom(dest).map_err(|e|
-            Error::with_cause(ErrorKind::Unavailable, "OsRng failed", e))
+        getrandom(dest).map_err(|e| {
+            #[cfg(feature="std")] {
+                rand_core::Error::from_cause(e)
```

Comment:
It would be nicer not to have to use a `cfg` here (e.g. make `from_code` work either way). That means the `std` feature is also not needed on this crate.